### PR TITLE
feat (kontena) Support for namespaced kontena stacks

### PR DESCRIPTION
**Pull request by aom**
Test without namespace:

```
$ echo "stack: foo-bar" | rev | cut -d' ' -f1 | cut -d'/' -f1 | rev
foo-bar
```

With namespace:

```
$ echo "stack: foo/bar" | rev | cut -d' ' -f1 | cut -d'/' -f1 | rev
bar
```

### CODE REVIEW

**Comment by matti** [comment]
instead of dropping the namespace, shouldn't it merge it?

this would allow installs of `jakolehm/stolon` and `matti/stolon` - otherwise they will conflict since the name `stolon` is already in use.

**Comment by matti** [comment]
the `$stack_name` (install name) should become `jakolehm/stolon` or `jakolehm-stolon` if the former doens't work.

**Comment by aom** [comment]
*TL;DR; I'll change the behaviour to merging the namespace with a dash. I'll leave it to you to decide whether to have it or not.*

I have installed a local stack from *kontena.yml* that defines stack name with namespace. Like `jakolehm/stolon` in your example.

When running:

```
kontena stack install
```

It also strips the namespace off leaving me with a stack called `stolon`.

This PR would be in line with Kontena's own convention but I agree that it would conflict with remote stacks where you'd want to retain the namespace.

In this case local stack should not even have the namespace because it is not meant to be pushed to a public repository.

**Comment by matti** [comment]
agreed.
